Makefile

```diff
@@ -6,85 +6,107 @@
 # Dependencies:
 # * make
 # * which
+# * npm
 # * curl: used if phpunit and composer are not installed to fetch them from the web
 # * tar: for building the archive
 
-app_name=$(notdir $(CURDIR))
-build_tools_directory=$(CURDIR)/build/tools
-build_source_directory=$(CURDIR)/build/source
-appstore_build_directory=$(CURDIR)/build/artifacts/appstore
-appstore_package_name=$(appstore_build_directory)/$(app_name)
-nc_cert_directory=$(HOME)/.nextcloud/certificates
+app_name=polls
+
+project_dir=$(CURDIR)
+build_dir=$(CURDIR)/build
+build_tools_dir=$(build_dir)/tools
+build_source_dir=$(build_dir)/source
+appstore_build_dir=$(build_dir)/artifacts/appstore
+appstore_package_name=$(appstore_build_dir)/$(app_name)
+nc_cert_dir=$(HOME)/.nextcloud/certificates
 composer=$(shell which composer 2> /dev/null)
 
-all: composer
+all: dev-setup lint build-js-production
+
+# Dev environment setup
+dev-setup: clean clean-dev npm-init composer
+
+npm-init:
+	npm install
+
+# Build js
+build-js-production:
+	npm run build
 
 # Installs and updates the composer dependencies. If composer is not installed
 # a copy is fetched from the web
 .PHONY: composer
 composer:
 ifeq (,$(composer))
 	@echo "No composer command available, downloading a copy from the web"
-	mkdir -p $(build_tools_directory)
+	mkdir -p $(build_tools_dir)
 	curl -sS https://getcomposer.org/installer | php
-	mv composer.phar $(build_tools_directory)
-	php $(build_tools_directory)/composer.phar install --prefer-dist
-	php $(build_tools_directory)/composer.phar update --prefer-dist
+	mv composer.phar $(build_tools_dir)
+	php $(build_tools_dir)/composer.phar install --prefer-dist
+	php $(build_tools_dir)/composer.phar update --prefer-dist
 else
 	composer install --prefer-dist
 	composer update --prefer-dist
 endif
 
-# Removes the appstore build
+# Lint
+lint:
+	npm run lint
+
+lint-fix:
+	npm run lint:fix
+
+# Removes the appstore build and compiled js files
 .PHONY: clean
 clean:
-	rm -rf ./build
+	rm -rf $(build_dir)
+	rm -f ./vendor
```

Comment:
Either use `-rf` since vendor is a folder or remove that line. I don't see a point, why the php deps should be deleted?

Reply:
Good question. I thought it is a good idea to make a clean dev environment and pull the fresh libraries. Anyway the composer seems not make sense, because the tests fail on my instance due to a missing base.php. Your decision. :-)

Reply:
I'd drop the `clean` target from the `appstore` target, because all files/folders in `clean` are regenerated in `appstore`. If anyone wants a clean repo, they can run `make clean && make appstore`.

Reply:
Hmmm. Maybe we should change the building commands another way: 

see last commit

In the end 
- `clean-dev:` prepares a fresh `dev-setup`
- `clean:` prepares for a new `appstore` build without the need for a complete new `dev-setup`
- `appstore:` just runs a new appstore build with compiling js
- `all:` runs `dev-setup` and `appstore` 

What do you mean

Reply:
@v1r0x Any more comments on this, or may I merge?

Reply:
Sorry, everything's fine :)